.grit/patterns/python/jwt_go_none_algorithm.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Detected use of the `'none'` algorithm in a JWT token and Instead, use an algorithm such as `'HS256'`
+tags: [fix, correctness, jwt, security]
+---
+
+Detected use of the `'none'` algorithm in a JWT token. The `'none'` algorithm assumes the integrity of the token has already been verified. This would allow a malicious actor to forge a `JWT` token that will automatically be verified. Do not explicitly use the `'none'` algorithm. Instead, use an algorithm such as `'HS256'`.
+
+## references
+- [Cryptographic Failures](https://owasp.org/Top10/A02_2021-Cryptographic_Failures)
+
+```grit
+engine marzano(0.1)
+language python
+
+or {`jwt.encode($params)`, `jwt.decode($params)`} where {
+    any {
+        $params <: contains `algorithm='none'` => `algorithm='HS256'`,
+        $params <: contains `algorithms=[$algo]` where {
```

Comment:
If it's an array of algorithms, we probably want to:

- Remove `'none'`
- (If and only), `none` is the only agoritmh, add 'HS256'

Please add and handle test cases for:
- `algorithms=["none", "other", "HS256"]`
- `algorithms=["HS256"]`
- `algorithms=["none", "md5"]`

Reply:
Hi @morgante, updated it now

Reply:
Also test this case: `algorithms=["none"]`. It should be replaced with HS256 if it's the only one.

Reply:
have been added [here](https://github.com/getgrit/stdlib/blob/2249eec42be76dc82703bb9356b3568093e72d08/.grit/patterns/python/jwt_go_none_algorithm.md?plain=1#L44)